lib/ecstatic.js

```diff
@@ -330,6 +339,19 @@ var ecstatic = module.exports = function (dir, options) {
 ecstatic.version = version;
 ecstatic.showDir = showDir;
 
+function hasGzipId12(gzipped, cb) {
+  var stream = fs.createReadStream(gzipped, { start: 0, end: 1, autoClose: true }),
+      isGzip = false;
+
+  stream.on("close", function () {
+    cb(isGzip);
+  });
+
+  stream.on("data", function (chunk) {
```

Comment:
Possible bug for a stream only giving one byte in the chunk? Unlikely, but I don't think fs makes any guarantees about this.

I've used concat-stream for this kind of thing in the past, it might be helpful for you here.